### PR TITLE
Fixes for issues #96 and #105 

### PR DESCRIPTION
**Pull request by djtfmartin**
Fix for AtlasOfLivingAustralia/la-pipelines#96 - making the initialisation of the centre points objects and vocab more thread safe by initialising in the transform.

Temporary fix for AtlasOfLivingAustralia/la-pipelines#105 while we investigate the resource cleanup issues.

### CODE REVIEW

File: livingatlas/pipelines/src/main/java/au/org/ala/pipelines/interpreters/ALALocationInterpreter.java [hunk 1]
```diff
@@ -94,124 +91,54 @@
...
-                    lr.getDecimalLongitude(),
-                    lr.getStateProvince());
-              }
+          if (stateProvinceCentrePoints.coordinatesMatchCentre(
+              lr.getStateProvince(), lr.getDecimalLatitude(), lr.getDecimalLongitude())) {
+            addIssue(lr, ALAOccurrenceIssue.COORDINATES_CENTRE_OF_STATEPROVINCE.name());
+          } else {
+            if (log.isTraceEnabled()) {
+              log.trace(
+                  "{},{} is not the centre of {}!",
```
Comment (timrobertson100) [inline livingatlas/pipelines/src/main/java/au/org/ala/pipelines/interpreters/ALALocationInterpreter.java:137]: Has this got a typo? Should it read "...is [at] the centre of {}" ?

Comment (djtfmartin) [reply livingatlas/pipelines/src/main/java/au/org/ala/pipelines/interpreters/ALALocationInterpreter.java:137]: it has yes - good catch

Comment (djtfmartin) [reply livingatlas/pipelines/src/main/java/au/org/ala/pipelines/interpreters/ALALocationInterpreter.java]: i.e. no issue needs to be raise....

Comment (timrobertson100) [reply livingatlas/pipelines/src/main/java/au/org/ala/pipelines/interpreters/ALALocationInterpreter.java]: Indeed. I am blind

File: livingatlas/pipelines/src/main/java/au/org/ala/pipelines/transforms/LocationTransform.java [hunk 2]
```diff
@@ -79,6 +84,17 @@ public void setup() {
...
     }
+
+    try {
+      countryCentrePoints = CountryCentrePoints.getInstance(alaConfig.getLocationInfoConfig());
+      stateProvinceCentrePoints =
+          StateProvinceCentrePoints.getInstance(alaConfig.getLocationInfoConfig());
+      stateProvinceVocab =
+          StateProvince.getInstance(alaConfig.getLocationInfoConfig().getStateProvinceNamesFile());
+    } catch (Exception e) {
+      log.error(e.getMessage(), e);
```
Comment (qifeng-bai) [inline livingatlas/pipelines/src/main/java/au/org/ala/pipelines/transforms/LocationTransform.java:95]: Shall we provide more hints to help users correct configuration file?

Comment (djtfmartin) [reply livingatlas/pipelines/src/main/java/au/org/ala/pipelines/transforms/LocationTransform.java:95]: @qifeng-bai Im wondering if we need to do this now, as by default the vocab files that are shipped in the JAR will be used unless the user has configured some externally via YAML.

I think when we originally added this help, the vocab files would only be loaded externally so we needed the error messages.

Comment (qifeng-bai) [reply livingatlas/pipelines/src/main/java/au/org/ala/pipelines/transforms/LocationTransform.java:96]: I am OK without hints at this moment.

**Pull request merged**